### ISSUE TITLE
Can withInfo display flow types as opposed to PropTypes?

### ISSUE THREAD

**Issue by jimjeffers** [storybooks/storybook#3017]
### Issue details

I migrated most of the my projects to utilize FlowKS instead of defining prop types. Is there already a best practice or solution to automatically display the Component<Props,State> defined on the component?

### Steps to reproduce

Create a component utilizing FlowJS. Example:

```js
class Button extends React.Component<{label: string, onClick: (e: MouseEvent)=> any}, {clickCount: number}> {
  handleClick(e: MouseEvent) {
    this.props.onClick(e)
    this.setState({clickCount: this.state.clickCount+1})
  }

  render() {
    return (<button onClick={this.handleClick}>{this.props.label}</button>)
  }
}
```

### Please specify which version of Storybook and optionally any affected addons that you're running

```json
{
    "@storybook/addon-actions": "^3.3.13",
    "@storybook/addon-info": "^3.3.13",
    "@storybook/addon-links": "^3.3.13",
    "@storybook/addon-notes": "^3.3.13",
    "@storybook/addons": "^3.3.13",
    "@storybook/react": "^3.3.13"
}
```

### Affected platforms 

```json
{
    "flow-bin": "^0.60.1",
    "react": "^16.2.0",
}
```

### Screenshots / Screencast / Code Snippets (Optional)

```html
<div>
  <h1 style="margin: 20px 0px 0px; padding: 0px 0px 5px; font-size: 25px; border-bottom-width: 1px; border-bottom-style: solid; border-bottom-color: rgb(238, 238, 238);">Prop Types</h1>
  <div>
    <h2 style="margin: 20px 0px 0px;">"Button" Component</h2>
    <small>No propTypes defined!</small>
  </div>
</div>
```

**Comment by Hypnosphi** [comment]
Weird, addon-info should support Flow out of the box. 
Can you please create a GitHub repo with minimal reproduction of your issue?

**Comment by jimjeffers** [comment]
@Hypnosphi Odd. You are right I just created a new project with CRA and flow works just fine for the proptypes. I'll continue to investigate what is causing it not to work in my specific app and report the cause ASAP.

**Comment by jimjeffers** [comment]
@Hypnosphi I deduced it to two helper functions I had in the same file as my component:

```js
const shortHand = (value: number): string =>
  value >= 1000 ? `${Math.round(value / 1000.0)}K` : `${value}`

const formatValue = (value: number, unit: Unit): string => {
  ...
  return `${value}`
}
```

I found the props would show up so long as I didn't have a helper method that accepted more than one parameter. Changing either helper to have two or more params caused the props to disappear. Quite strange! 

I moved these helpers into a separate module and the props now show up in storybook. Probably a bug but the work around should be noted somewhere. Unless it is and I just didn't pay attention!
